fe/fe-core/src/main/java/org/apache/doris/statistics/util/Connection.java

```diff
@@ -0,0 +1,424 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.statistics.util;
+
+import org.apache.doris.common.Config;
+import org.apache.doris.mysql.MysqlCapability;
+import org.apache.doris.mysql.MysqlCommand;
+import org.apache.doris.mysql.privilege.PaloAuth;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class Connection implements AutoCloseable {
+    private static final Logger LOG = LogManager.getLogger(Connection.class);
+
+    private static final int BUFFER_SIZE = 1024;
+    private static final int ERROR_STATUS = 0xff;
+    private static final int OK_STATUS = 0x00;
+
+    private final StringBuilder sb = new StringBuilder();
+
+    private String host;
+    private int port;
+    private String username;
+    private String password;
+    private String database;
+
+    private Socket socket;
+
+    private InputStream in;
+    private OutputStream out;
+
+    private ByteBuffer hpBuffer;
+    private ByteBuffer buffer;
+
+    private HandshakePacket handshakePacket;
+
+    private Boolean initialized = false;
+
+    public Connection() {
+    }
+
+    /**
+     * Fe query be does not require a password,
+     * root and admin user is allowed to login from 127.0.0.1, in case user forget password.
+     *
+     * @param database database name
+     * @see PaloAuth
+     */
+    public Connection(String database) {
+        host = "127.0.0.1";
+        port = Config.query_port;
+        username = PaloAuth.ROOT_USER;
+        password = "";
```

Comment:
be's password is same with fe's password? should we get correct password automatically rather than set by user?